starter-code/iteration-3/main.ts

```diff
@@ -1,8 +1,59 @@
 // Add the reference to the interface
+import {TodoItemInterface, TodoListInterface} from './interfaces'
 
 // Create class TodoItem that implements the corresponding interface
+class TodoItem implements TodoItemInterface{
+  title:string;
+  done:boolean = false;
+  updatedAt:Date;
+
+  constructor(title:string){
+    this.title = title;
+    this.updatedAt = new Date();
+  }
+
+  toggleStatus(){
+    this.done = !this.done;
+    this.updatedAt = new Date();
+  }
+}
 
 // Create class TodoList that implements the corresponding interface
+class TodoList implements TodoListInterface{
+  todos:Array<TodoItemInterface> = [];
+
+  addTask(item:TodoItemInterface){
+    this.todos.push(item);
+    console.log("Task: " + item.title + " inserted in the list");
+    return this.todos.length;
+  }
+  
+  listAllTasks():void{
+    for(var i =0; i<this.todos.length; i++){
+      console.log(this.todos[i].title);
+    }
+  }
+
+  deleteTask(item:TodoItemInterface):number{
+    var position = this.todos.indexOf(item);
+    if(position === -1){
```

Comment:
you could have used a splice here as well:
`    this.tasks.splice(index, 1);
`